server/config/service.go

```diff
@@ -107,37 +220,26 @@ func (s *service) reloadMattermostConfig() *model.Config {
 	return mmconf
 }
 
-func (s *service) Reconfigure(stored StoredConfig, log utils.Logger, services ...Configurable) error {
-	mmconf := s.reloadMattermostConfig()
-	newConfig := s.Get()
-
-	// GetLicense silently drops an RPC error
-	// (https://github.com/mattermost/mattermost-server/blob/fc75b72bbabf7fabfad24b9e1e4c321ca9b9b7f1/plugin/client_rpc_generated.go#L864).
-	// When running in Mattermost cloud we must not fall back to the on-prem mode, so in case we get a nil retry once.
-	license := s.mm.System.GetLicense()
-	if license == nil {
-		license = s.mm.System.GetLicense()
-		if license == nil {
-			log.Debugf("failed to fetch license twice. May incorrectly default to on-prem mode")
-		}
+func (s *service) Reconfigure(newStoredConfig StoredConfig, quiet bool, services ...Configurable) error {
+	var configLogger utils.Logger = utils.NilLogger{}
+	if !quiet {
+		configLogger = s.NewBaseLogger()
```

Comment:
When this line gets called, `s.mattermostConfig` has not been refreshed. This means a change in the dev mode does not get picked up. Also, the logger used in `newInitializedConfig` will not use the updated `LogConfig`.

Reply:
`configLogger` is the logger to use while handling the configuration change itself, so it is appropriate that it uses the "old" configuration for itself. Also, FWIW, it is set to a "real" value only when called the first time in `OnActivate`, afterwards it's always `nil`.

All other config changes should be picked up by `newInitializedConfig`, or am I missing something else?

Reply:
By changing the code to pass a `Logger` to `Reconfigure` the bug has been fixed. :+1: